addons/addon-base-raas/packages/base-raas-services/lib/data-source/data-source-reachability-service.js

```diff
@@ -0,0 +1,262 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const _ = require('lodash');
+const Service = require('@aws-ee/base-services-container/lib/service');
+const { allowIfActive, allowIfAdmin } = require('@aws-ee/base-services/lib/authorization/authorization-utils');
+const { processInBatches } = require('@aws-ee/base-services/lib/helpers/utils');
+
+const workflowIds = {
+  bulkCheck: 'wf-bulk-reachability-check',
+  accountStatusChange: 'wf-ds-account-status-change',
+};
+
+class DataSourceReachabilityService extends Service {
+  constructor() {
+    super();
+    this.dependency([
+      'auditWriterService',
+      'dataSourceAccountService',
+      'studyService',
+      'workflowTriggerService',
+      'aws',
+      'roles-only/applicationRoleService',
+    ]);
+  }
+
+  async bulkReach(requestContext, { status }, { forceCheckAll = false } = {}) {
+    status = status || '*';
+    const workflowTriggerService = await this.service('workflowTriggerService');
+    const dsAccountIds = await this._getDsAccountsWithStatus(requestContext, status);
+
+    if (!_.isEmpty(dsAccountIds)) {
+      await workflowTriggerService.triggerWorkflow(
+        requestContext,
+        { workflowId: workflowIds.bulkCheck },
+        {
+          requestContext,
+          status,
+          dsAccountIds,
```

Comment:
You might want to leave a TODO comment here stating that if we have hundreds of account, we need to store the account ids in an s3 object and pass the s3 url to the workflow and the workflow step will have to get the object from S3.  This is because a workflow payload has a size limit of 32k.

Reply:
For now, I'm adding a guard to not trigger a workflow if number of DS Accounts needing a check exceeds 1000.